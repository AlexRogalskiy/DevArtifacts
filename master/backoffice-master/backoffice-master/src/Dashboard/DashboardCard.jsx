import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import {
  Card,
  CardContent,
  Typography,
  withStyles,
  Avatar,
} from '@material-ui/core'

import DisabledIcon from '@material-ui/icons/Lock'

const styles = theme => ({
  root: {
    cursor: 'pointer',
  },
  content: {
    paddingBottom: `${theme.spacing.unit * 2}px !important`,
  },
  avatar: {
    float: 'left',
    marginRight: theme.spacing.unit * 2,
    backgroundColor: theme.palette.secondary.dark,
  },
  title: {
    marginBottom: 0,
  },
  disabled: {
    opacity: 0.75,
    pointerEvents: 'none',
    filter: 'grayscale(30%)',
  },
  icon: {
    float: 'right',
    opacity: 0.5,
  },
})

const DashboardCard = ({
  title,
  description,
  handleClick,
  icon: Icon,
  isDisabled,
  classes,
}) => {
  const rootClasses = classNames({
    [classes.root]: true,
    [classes.disabled]: isDisabled,
  })

  return (
    <Card onClick={handleClick} className={rootClasses}>
      <CardContent className={classes.content}>
        {isDisabled ? (
          <DisabledIcon className={classes.icon} />
        ) : null}

        {Icon ? (
          <Avatar className={classes.avatar}>
            <Icon />
          </Avatar>
        ) : null}

        <Typography variant="headline" className={classes.title} component="h2">
          {title}
        </Typography>

        {description ? (
          <Typography variant="body1">{description}</Typography>
        ) : null}
      </CardContent>
    </Card>
  )
}

DashboardCard.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  handleClick: PropTypes.func.isRequired,
  icon: PropTypes.func,
  isDisabled: PropTypes.bool,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
}

DashboardCard.defaultProps = {
  description: null,
  icon: null,
  isDisabled: false,
}

export default withStyles(styles)(DashboardCard)
