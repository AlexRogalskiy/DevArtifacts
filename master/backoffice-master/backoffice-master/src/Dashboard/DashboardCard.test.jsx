import React from 'react'
import { mount } from 'enzyme'

import DashboardCard from './DashboardCard'

it('renders correctly', () => {
  const tree = mount(<DashboardCard
    title="Title"
    description="desc"
    handleClick={() => { }}
  />)

  expect(tree).toMatchSnapshot()
})


it('renders disabled', () => {
  const tree = mount((
    <DashboardCard
      title="Title"
      description="desc"
      handleClick={() => { }}
      isDisabled
    />
  ))


  expect(tree).toMatchSnapshot()
})
