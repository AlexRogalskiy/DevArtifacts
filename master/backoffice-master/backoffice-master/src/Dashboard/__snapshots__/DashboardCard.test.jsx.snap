// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root-8 MuiPaper-elevation1-11 MuiPaper-rounded-9 MuiCard-root-7 DashboardCard-root-1"
  onClick={[Function]}
>
  <div
    className="MuiCardContent-root-35 DashboardCard-content-2"
  >
    <h2
      className="MuiTypography-root-36 MuiTypography-headline-41 DashboardCard-title-4"
    >
      Title
    </h2>
    <p
      className="MuiTypography-root-36 MuiTypography-body1-45"
    >
      desc
    </p>
  </div>
</div>
`;

exports[`renders disabled 1`] = `
<div
  className="MuiPaper-root-8 MuiPaper-elevation1-11 MuiPaper-rounded-9 MuiCard-root-7 DashboardCard-root-1 DashboardCard-disabled-5"
  onClick={[Function]}
>
  <div
    className="MuiCardContent-root-35 DashboardCard-content-2"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-61 DashboardCard-icon-6"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
      />
    </svg>
    <h2
      className="MuiTypography-root-36 MuiTypography-headline-41 DashboardCard-title-4"
    >
      Title
    </h2>
    <p
      className="MuiTypography-root-36 MuiTypography-body1-45"
    >
      desc
    </p>
  </div>
</div>
`;
